javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,185 @@
+
+  override def toArray[F <: AnyRef](array: Array[F]): Array[F] =
+    if (_size > array.length) {
+      Arrays.copyOf(values.asInstanceOf[Array[E with AnyRef]],
+                    _size,
+                    array.getClass.asInstanceOf[Class[_ <: Array[F]]])
+    } else {
+      System.arraycopy(values, 0, array, 0, _size)
+      if (array.length > _size)
+        array(_size) = null.asInstanceOf[F] // see tests
+      array
+    }
+
+  def get(index: Int): E = {
+    checkIndexInBounds(index)
+    values(index).asInstanceOf[E]
+  }
+
+  override def set(index: Int, element: E): E = {
+    val e = get(index)
+    values(index) = element.asInstanceOf[AnyRef]
+    e
+  }
+
+  override def add(e: E): Boolean = {
+    values(_size) = e.asInstanceOf[AnyRef]
+    _size += 1
+    true
+  }
+
+  override def add(index: Int, element: E): Unit = {
+    checkIndexOnBounds(index)
+    System.arraycopy(values, index, values, index + 1, size - index)
+    values(index) = element.asInstanceOf[AnyRef]
+    _size += 1
+  }
+
+  override def remove(index: Int): E = {
+    checkIndexInBounds(index)
+    val elem = values(index)
+    remove0(index)
+    elem.asInstanceOf[E]
+  }
+
+  override def remove(that: Any): Boolean = {
+    var i     = 0
+    var found = false
+    while (i < _size && !found) {
+      if (values(i) equals that) {
+        found = true
+        remove0(i)
+      }
+      i += 1
+    }
+    found
+  }
+
+  // Shared code between different removes.
+  private def remove0(index: Int): Unit = {
+    // if it is the last element, we don't need to move any elements behind it!
+    if (index - 1 != _size)
+      System.arraycopy(values, index + 1, values, index, _size - index)
+    _size -= 1
+    values(_size) = null
+  }
+
+  // Removes all values, but doesn't reallocate or resize the underlying array.
+  override def clear(): Unit = {
+    var i = 0
+    while (i < _size) {
+      values(i) = null
+      i += 1
+    }
+    _size = 0
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    c match {
+      case other: ArrayList[_] =>
+        val otherSize = other.size
+        if (otherSize > 0) {
+          ensureCapacity(_size + otherSize)
+          System.arraycopy(other.values, 0, values, _size, otherSize)
+          true
+        } else {
+          false
+        }
+      case _ => super.addAll(c)
+    }
+  }
+
+  override def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
+    c match {
+      case other: ArrayList[_] =>
+        val otherSize = other.size
+        if (otherSize > 0) {
+          ensureCapacity(_size + otherSize)
+          System
+            .arraycopy(values, index, values, otherSize + index, _size - index)
+          System.arraycopy(other.values, 0, values, index, otherSize)
+          true
+        } else {
+          false
+        }
+      case _ => super.addAll(index, c)
+    }
+  }
+
+  // Needed to avoid linking error
```

Comment:
I've just filed an issue for this: #375 

Can you please change the message `Workaround for #375` ?